solutions/instances/main.tf

```diff
@@ -191,7 +193,7 @@ module "cos_instance" {
   providers = {
     ibm = ibm.cos
   }
-  count                    = (var.existing_cos_instance_crn == null) ? 1 : 0 # no need to call COS module if consumer is using existing COS instance
+  count                    = (var.existing_cos_instance_crn == null) && (var.log_analysis_provision == true) && (var.configure_cos_target_and_route_for_atracker_events == true) ? 1 : 0 # no need to call COS module if consumer is using existing COS instance
```

Comment:
why the following condition is required?
`(var.log_analysis_provision == true) && (var.configure_cos_target_and_route_for_atracker_events == true)`